lib/ledger_sync/resource_attribute_set.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module LedgerSync
+  class ResourceAttributeSet
+    attr_reader :attributes
+
+    def initialize(attributes)
+      @attributes = attributes
+    end
+
+    def types
+      @types ||= self.class.types
+    end
+
+    def valid?
+      return true if value.nil?
+      return true if valid_classes.select { |e| value.is_a?(e) }.any?
+
+      false
+    end
+
+    def valid_classes
+      types[type]
+    end
+
+    def self.types
+      @types ||= TYPES
```

Comment:
should this be `@@types` since it's on the class?

Reply:
Yup.  And this probably explains a lot of why I wrote things the way I did...lol